internal/common/testing/container/container_integration_test.go

```diff
@@ -0,0 +1,61 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build integration
+
+package container
+
+import (
+	"context"
+	"net/http"
+	"testing"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/client"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestContainerIntegration(t *testing.T) {
+	cli, err := client.NewClientWithOpts(client.FromEnv, client.WithAPIVersionNegotiation())
+	if err != nil {
+		t.Fatalf("failed creating docker client from environment: %v", err)
+	}
+	defer cli.Close()
+
+	// Remove image so more code paths are hit. (Don't need to prune children layers though to keep test faster.)
+	_, _ = cli.ImageRemove(context.Background(), "docker.io/library/nginx:1.17", types.ImageRemoveOptions{
+		Force:         true,
+		PruneChildren: false,
+	})
+
+	con := New(t)
+	started := con.StartImage("docker.io/library/nginx:1.17", WithPortReady(80))
+
+	require.NotEmpty(t, started.AddrForPort(80), "IP address was empty")
+	require.NotEmpty(t, started.ID)
+
+	resp, err := http.Get("http://" + started.AddrForPort(80))
+	require.NoError(t, err)
+	resp.Body.Close()
+	require.Equal(t, resp.StatusCode, http.StatusOK)
+
+	con.Cleanup()
+
+	assert.Nil(t, con.runningContainers, "started containers should be empty after cleanup")
```

Comment:
Would you like to use the corresponding method from the `require` module instead? So we don't need to import both `require` and `assert` modules?

Reply:
require and assert have different behaviors. Usually you use require if you want the test to fail immediately whereas asserts continue execution of the test so that additional assertions can run.